inc/inventorycomputerinventory.class.php

```diff
@@ -412,7 +412,18 @@ function sendCriteria($p_DEVICEID, $arrayinventory) {
          }
          $inputdb['rules_id'] = $data['_ruleid'];
          $inputdb['method'] = 'inventory';
-         $pfIgnoredimportdevice->add($inputdb);
+         $inputdb['plugin_fusioninventory_agents_id'] = $_SESSION['plugin_fusioninventory_agents_id'];
+
+         // if existing ignored device, update it
+         if ($found = $pfIgnoredimportdevice->find("`plugin_fusioninventory_agents_id` =
```

Comment:
problem because `$inputdb['serial']` not exist in some cases (case the device not have serial number or have serial number blacklisted)